firecrest/v2/_async/Client.py

```diff
@@ -114,26 +114,27 @@ async def chunk_reader(f, c):
 
                 yield data
 
-        self._client.log(
-            logging.DEBUG,
-            f"Uploading part {index + 1} to {url}"
-        )
         start = index * chunk_size
         if start + chunk_size > self._total_file_size:
             content_length = self._total_file_size - start
         else:
             content_length = chunk_size
 
-        async with aiofiles.open(self._local_file, "rb") as f:
-            await f.seek(start)
-            resp = await self._client._session.put(
-                url=url,
-                content=chunk_reader(f, self._chunk_size),
-                timeout=None,
-                headers={
-                    "Content-Length": str(content_length)
-                }
+        async with self._client._upload_semaphone:
```

Comment:
This is a typo here, right?
```suggestion
        async with self._client._upload_semaphore:
```